# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/wgpu_bindings/src/server.rs
# Commit: 1b7954a266d5
# Full Hash: 1b7954a266d5f9cd4d75b73857f253de3abf26f9
# Author: Erich Gubler <erichdongubler@gmail.com>
# Date: 2025-01-18 09:13:14
# Description:
#   Bug 1939731 - fix(webgpu): populate texture resource registry on external texture validation failure r=webgpu-reviewers,nical
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D234587
# ==============================================================================

diff -r 44e482539c57 -r 1b7954a266d5 gfx/wgpu_bindings/src/server.rs
--- a/gfx/wgpu_bindings/src/server.rs	Fri Jan 17 20:22:41 2025 +0000
+++ b/gfx/wgpu_bindings/src/server.rs	Fri Jan 17 20:25:41 2025 +0000
@@ -1967,6 +1967,7 @@
                     if desc.size.width > limits.max_texture_dimension_2d
                         || desc.size.height > limits.max_texture_dimension_2d
                     {
+                        self.create_texture_error(Some(id), &desc);
                         error_buf.init(ErrMsg {
                             message: "size exceeds limits.max_texture_dimension_2d",
                             r#type: ErrorBufferType::Validation,
@@ -1979,6 +1980,7 @@
                         && desc.usage.contains(wgt::TextureUsages::STORAGE_BINDING)
                         && !features.contains(wgt::Features::BGRA8UNORM_STORAGE)
                     {
+                        self.create_texture_error(Some(id), &desc);
                         error_buf.init(ErrMsg {
                             message: "Bgra8Unorm with GPUStorageBinding usage with BGRA8UNORM_STORAGE disabled",
                             r#type: ErrorBufferType::Validation,
